src/s3ql/fs.py

```diff
@@ -932,9 +943,16 @@ async def open(self, id_, flags, ctx):
             and (self.failsafe or self.inodes[id_].locked)):
             raise FUSEError(errno.EPERM)
 
-        if flags & os.O_TRUNC:
-            log.warning('open() with O_TRUNC is not yet supported.')
-            raise FUSEError(errno.ENOTSUP)
+        if (flags & os.O_RDWR or flags & os.O_WRONLY) and flags & os.O_TRUNC:
+            try:
+                attr = await self.getattr(id_, ctx)
+                if not stat.S_ISREG(attr.st_mode):
+                    log.warning('open() with O_TRUNC is only supported for regular files.')
+                    raise FUSEError(errno.ENOTSUP)
+                attr.st_size = 0
+                await self.setattr(id_, attr, _UpdateSize(), None, ctx)
+            except KeyError:
```

Comment:
Hmm. Could you explain what case this is supposed to catch? Note that `open()` is guaranteed to be called for an existing file (it does not receive a `name` argument, so it can't create a new file).

Reply:
I suspected that's the case but just wanted to be on the safe side and double check. I removed the `KeyError` check.